managers/views.py

```diff
@@ -160,3 +165,47 @@ def delete(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpRespons
         response = super().delete(request, args, kwargs)
         logger.info(f"Project with id: {kwargs['pk']} has been deleted by user with id: {self.request.user.pk}")
         return response
+
+
+@method_decorator(login_required, name="dispatch")
+@method_decorator(
+    check_permissions(allowed_user_types=[CustomUser.UserType.ADMIN.name, CustomUser.UserType.MANAGER.name]),
+    name="dispatch",
+)
+class ManageTaskActivitiesInProjectView(FormView, UserIsManagerOfCurrentProjectMixin, SingleObjectMixin):
+    model = Project
+    template_name = "managers/project_task_activities.html"
+    form_class = TaskActivityForm
+
+    def _delete_task_activity_from_relation(self) -> None:
+        task_activity = get_object_or_404(TaskActivityType, pk=self.request.GET.get("task_activity"))
+        task_activity.projects.remove(self.object)
+
+    def _add_task_activities_to_relation(self, list_of_task_activities: list) -> None:
+        for task_activity_id in list_of_task_activities:
+            task_activity = get_object_or_404(TaskActivityType, pk=task_activity_id)
+            task_activity.projects.add(self.object)
```

Comment:
Same here, you can query for all `TaskActivityType` in single request, and set result using `set()`.